Emby.Server.Implementations/EntryPoints/ExternalPortForwarding.cs

```diff
@@ -153,50 +145,46 @@ private async void CreateRules(INatDevice device)
 
             // On some systems the device discovered event seems to fire repeatedly
             // This check will help ensure we're not trying to port map the same device over and over
-            var address = device.DeviceEndpoint;
-
-            lock (_createdRulesLock)
+            if (!_createdRules.TryAdd(device.DeviceEndpoint, 0))
             {
-                if (!_createdRules.Contains(address))
-                {
-                    _createdRules.Add(address);
-                }
-                else
-                {
-                    return;
-                }
+                return Task.CompletedTask;
             }
 
-            try
-            {
-                await CreatePortMap(device, _appHost.HttpPort, _config.Configuration.PublicPort).ConfigureAwait(false);
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError(ex, "Error creating http port map");
-                return;
-            }
+            return Task.WhenAll(CreatePortMaps(device));
+        }
 
-            try
-            {
-                await CreatePortMap(device, _appHost.HttpsPort, _config.Configuration.PublicHttpsPort).ConfigureAwait(false);
-            }
-            catch (Exception ex)
+        private IEnumerable<Task> CreatePortMaps(INatDevice device)
+        {
+            yield return CreatePortMap(device, _appHost.HttpPort, _config.Configuration.PublicPort);
```

Comment:
I would rather we _not_ expose plain http port unless user explicitly told us to...

Reply:
That is a good point, I'd say you're right. I was just copying the existing functionality here.

This would require a new configuration option and would be a breaking change though, maybe it should go in a separate PR?

Reply:
Sounds good to me, yeah. Atomic PRs FTW!

Reply:
Created https://github.com/jellyfin/jellyfin/issues/2901 to address this